runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MetricsContainerImpl.java

```diff
@@ -187,6 +186,44 @@ private MonitoringInfo counterUpdateToMonitoringInfo(MetricUpdate<Long> metricUp
     return builder.build();
   }
 
+  /**
+   * @param metricUpdate
+   * @return The MonitoringInfo generated from the metricUpdate.
+   */
+  @Nullable
+  private MonitoringInfo distributionUpdateToMonitoringInfo(
+      MetricUpdate<org.apache.beam.runners.core.metrics.DistributionData> metricUpdate) {
+    SimpleMonitoringInfoBuilder builder = new SimpleMonitoringInfoBuilder(true);
+    MetricName metricName = metricUpdate.getKey().metricName();
+    if (metricName instanceof MonitoringInfoMetricName) {
```

Comment:
For my edification: Metrics in this branch would be system metrics because they are created via URN rather than namespace+name, correct?

Reply:
Yes, correct